sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1680,6 +1681,62 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     checkAnswer(ds, Seq(Row("a"), Row("b"), Row("c")))
   }
 
+  test("SPARK-26837:Prune nested serializers") {
+    def testSerializer(df: DataFrame, structFields: Seq[Seq[String]]): Unit = {
+      val serializer = df.queryExecution.optimizedPlan.collect {
+        case s: SerializeFromObject => s
+      }.head
+
+      val structs = serializer.serializer(0).collect {
```

Comment:
`serializer(0)` -> `serializer.head`.

Reply:
I originally wanted to keep the test simple. But I think we should also test other serializers too. I updated the test case.